webapp/sources/rudder/rudder-web/src/main/elm/sources/ComplianceMode/DataTypes.elm

```diff
@@ -0,0 +1,28 @@
+module ComplianceMode.DataTypes exposing (..)
+
+import Http exposing (Error)
+import Http.Detailed
+
+--
+-- All our data types
+--
+
+type ComplianceMode = FullCompliance | ChangesOnly | ReportsDisabled | UnknownMode
```

Comment:
I don 't think you need an UnknownMode. It should make an error if the value is invalid.

Reply:
I've renamed UnknownMode to ErrorMode, but I've kept it to make it easier to handle this error case, by displaying a message in the form, preventing it from being saved and displaying an error notification.

Reply:
GReat !